sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,287 @@
+# Migrating from old to preview management SDK.
+
+There are several differences between the old sdk and this preview. Here's an example of how to create a Virtual Machine with both SDKs:
+
+## Create a Virtual Machine example
+
+### Import the namespaces
+#### Old (Microsoft.Azure.Management._)
+```C#
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Rest;
+using System;
+using System.Threading.Tasks;
+```
+#### New (Azure.Azure.ResourceManager._ Preview)
+```C#
+using Azure.Identity;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Compute;
+using Azure.ResourceManager.Network;
+using System;
+using System.Threading.Tasks;
+```
+
+### Setting up the clients
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials();
+var computeClient = new ComputeManagementClient(credentials);
+var networkClient = new NetworkManagementClient(credentials);
+var managedServiceIdentityClient = new ManagedServiceIdentityClient(credentials);
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+```
+As you can see, authentication is now handled by Azure.Identity, and now just a single client is needed, from which you can get the `DefaultSubscription` and start managing your resources. 
+
+### Create a Resource Group
+#### Old
+```C#
+ServiceClientCredentials credentials = getMyCredentials(); 
+var resourcesClient = new ResourceManagementClient(credentials);
+
+string rgName = "QuickStartRG";
+string location = "WestUS2";
+resourcesClient.ResourceGroups.CreateOrUpdate(
+    rgName,
+    new ResourceGroup
+    {
+        Location = location,
+        Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+    });
+```
+#### New
+```C#
+var armClient = new ArmClient(new DefaultAzureCredential());
+Subscription subscription = armClient.DefaultSubscription;
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+Location location = Location.WestUS2;
+string rgName = "QuickStartRG";
+ResourceGroup resourceGroup = await rgContainer.Construct(location).CreateOrUpdateAsync(rgName)
+```
+
+In the new libraries, the Resource Group is created trought a container that came from the subscription. Also, public locations are provided trought the `Location` object, but it can be specified as a string as well. Additionally, there's no need to create a `ResourceGroup` object to send parameters when creating a resource group, since tags can be added directly from the resource group variable and the location is specified in the `Construct()` method.
```

Comment:
```suggestion
In the new libraries, the Resource Group is created through a container that came from the subscription. Also, public locations are provided through the `Location` object, but can be specified using a string as well. Additionally, there's no need to create a `ResourceGroup` object to send parameters when creating a resource group, since tags can be added directly from the resource group variable and the location is specified in the `Construct()` method.
```

Reply:
Actually, we probably want to say a couple of things here a bit differently.

First, the main difference is that the legacy client api represents all operations as flat, while the preview client API represents the hierarchy of resources, so that, for example, you can use a 'subscriptionContainer' to manage the resources in a particular subscription, you can use a resourceGroupContainer to manage the resources in a particular resource group, etc.

Second, it isn't quite true that you don't have to create a resource group object - rather, we provide some 'Construct' method helpers in the resource group container to help create common configurations of common resources, like resource group.  These make it easier to discover the correct way to create properly configured resources.

Reply:
Third, we have provided some common classes to represent commonly-used constructs, like Location, and allow you to use them directly throughout the APIs, again, making it easier to discover how to properly configure resources.